### DISCUSSION TITLE
[@types/testing-library__jest-dom] Issue with `Matchers` declaration with `@types/testing-library__jest-dom` and `expect`

### DISCUSSION THREAD

**Discussion by benjdlambert** [DefinitelyTyped/DefinitelyTyped#65987] [Issues with a @types package]
Hey :wave: looks like the most recent release of types for `@testing-library/jest-dom` has broken our pipeline https://github.com/backstage/backstage/pull/18554 😢 

```
node_modules/@types/testing-library__jest-dom/index.d.ts:21:15 - error TS2428: All declarations of 'Matchers' must have identical type parameters.

21     interface Matchers<R = void, T = {}> extends TestingLibraryMatchers<typeof expect.stringContaining, R> {}
                 ~~~~~~~~

node_modules/expect/build/index.d.ts:107:26 - error TS2428: All declarations of 'Matchers' must have identical type parameters.

107 export declare interface Matchers<R extends void | Promise<void>, T = unknown> {
```

Not sure what the best fix is here?

[a comment by an automated account is not shown]

**Comment by Vinnl** [comment]
@benjdlambert Are you able to reproduce the issue locally? If so, could you open `node_modules/@types/testing-library__jest-dom/index.d.ts` and apply [these changes](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65991/files) (i.e. replace the first `{}` by `unknown` on line 21) manually, and report back whether that fixes the issue for you?

**Reply by G-Rath** [reply]
Doing that fixes it for me

**Reply by benjdlambert** [reply]
@Vinnl can confirm this works as expected :tada:

**Reply by Vinnl** [reply]
Thank you both! Then we'll only need to wait for an approval :)

**Comment by niftinessafoot** [comment]
It appears this got merged in today https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65991, but TypeScript is still unhappy, this time with the `R` param:

```
node_modules/@types/testing-library__jest-dom/index.d.ts:21:15 - error TS2428: All declarations of 'Matchers' must have identical type parameters.

21     interface Matchers<R = void, T = {}> extends TestingLibraryMatchers<typeof expect.stringContaining, R> {}
                 ~~~~~~~~

node_modules/expect/build/index.d.ts:107:26 - error TS2428: All declarations of 'Matchers' must have identical type parameters.

107 export declare interface Matchers<R extends void | Promise<void>, T = unknown> {
```

VSCode _wants_ `R extends void | Promise<void> = void`. Updating the declaration manually eliminates the error.
